zerver/lib/upload.py

```diff
@@ -149,6 +149,11 @@ def get_file_info(request, user_file):
     # type: (HttpRequest, File) -> Tuple[text_type, Optional[text_type]]
 
     uploaded_file_name = user_file.name
+    if not isinstance(user_file.name, str):
+        # This check makes sure that the type of the file name is not
+        # Python 2 Unicode.
+        uploaded_file_name = uploaded_file_name.encode('ascii')
```

Comment:
Nope, it seems this happens outside our code. I checked by creating a middleware and verifying the values over there. This middleware ran before all other middlewares and our view logic.